packages/account/src/Components/leave-confirm/leave-confirm.tsx

```diff
@@ -46,30 +46,36 @@ const LeaveConfirmMessage = ({ back, leave }: TLeaveConfirmMessage) => {
  */
 export const TransitionBlocker = ({ dirty, onDirty }: TTransitionBlocker) => {
     const [show, setShow] = React.useState(false);
-    const [next_location, setNextLocation] = React.useState<{ pathname: string } | null>(null);
     const history = useHistory();
     const { isMobile } = useDevice();
+    const [location, setNextLocation] = useLocation();
```

Comment:
```suggestion
    const location = useLocation();
```
i dont think location has these properties. it returns the `Location` object. You can know what properties does a specific code library by right clicking the method and going to the definition (see the screenshot).

```
export interface Location<S = LocationState> {
    pathname: Pathname;
    search: Search;
    state: S;
    hash: Hash;
    key?: LocationKey | undefined;
}
```

<img width="449" alt="Screenshot 2024-07-16 at 11 39 30 AM" src="https://github.com/user-attachments/assets/f2226b1d-093e-4fa6-ad40-fd75fc9bec4a">
ly returns a location object. you can inspect it by